pkg/charts/utils.go

```diff
@@ -197,6 +197,12 @@ func ComputeCalicoChartValues(
 		calicoChartValues["global"].(map[string]string)["snatToUpstreamDNSEnabled"] = strconv.FormatBool(snatToUpstreamDNS)
 	}
 
+	if config != nil && config.AutoScaling != nil && config.AutoScaling.AutoscalingMode == AutoscalingModeVPA && wantsVPA {
+		calicoChartValues["vpa"].(map[string]string)["enabled"] = true
+	} else {
+		calicoChartValues["vpa"].(map[string]string)["enabled"] = false
+	}
```

Comment:
On second thought, please do not reuse `vpa.enabled`, but add two additional fields under `vpa` for `calico-node` and `calico-typha`, e.g. `vpa.node` and `vpa.typha`. We should not disable VPA for `calico-kube-controllers` just because we use cluster-proportional autoscaling for `calico-node` and `calico-typha`.

Please add the new fields [here](https://github.com/gardener/gardener-extension-networking-calico/blob/00acf2dc10a336eea80c35f3c2fd4ca85a5d98d1/pkg/charts/utils.go#L165) and [here](https://github.com/gardener/gardener-extension-networking-calico/blob/00acf2dc10a336eea80c35f3c2fd4ca85a5d98d1/charts/internal/calico/values.yaml#L40). Afterwards, please use the fields in the corresponding resource templates [here](https://github.com/gardener/gardener-extension-networking-calico/tree/00acf2dc10a336eea80c35f3c2fd4ca85a5d98d1/charts/internal/calico/templates).

In addition to that, you most likely need to run `make generate` to generate some additional code.